File: easy/933. Number of Recent Calls/933. Number of Recent Calls.py
class RecentCounter:

    def __init__(self):
        self.requests = []
        self.prev_start = 0

    def ping(self, t: int) -> int:
        self.requests.append(t)
        time_range = t - 3000

        start = self.prev_start
        while self.requests[start] < time_range:
            start += 1

        self.prev_start = start
        return len(self.requests) - start


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: easy/933. Number of Recent Calls/933. Number of Recent Calls.go
package main

type RecentCounter struct {
	r []int
}

func Constructor() RecentCounter {
	return RecentCounter{}
}

func (this *RecentCounter) Ping(t int) int {
	this.r = append(this.r, t)
	for this.r[0] < t-3000 {
		this.r = this.r[1:]
	}
	return len(this.r)
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * obj := Constructor();
 * param_1 := obj.Ping(t);
 */
